gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -58,28 +56,27 @@
  * An implementation for {@link DagProc} that launches a new job.
  */
 @Slf4j
-@RequiredArgsConstructor
 public class LaunchDagProc extends DagProc<Optional<Dag<JobExecutionPlan>>, Optional<Dag<JobExecutionPlan>>> {
-  private final LaunchDagTask launchDagTask;
   private final FlowCompilationValidationHelper flowCompilationValidationHelper;
   // todo - this is not orchestration delay and should be renamed. keeping it the same because DagManager is also using
   // the same name
   private static final AtomicLong orchestrationDelayCounter = new AtomicLong(0);
+
+  public LaunchDagProc(LaunchDagTask launchDagTask, FlowCompilationValidationHelper flowCompilationValidationHelper) {
+    super(launchDagTask);
+    this.flowCompilationValidationHelper = flowCompilationValidationHelper;
+  }
+
   static {
     metricContext.register(
         metricContext.newContextAwareGauge(ServiceMetricNames.FLOW_ORCHESTRATION_DELAY, orchestrationDelayCounter::get));
   }
 
-  @Override
-  protected DagManager.DagId getDagId() {
-    return this.launchDagTask.getDagId();
-  }
-
   @Override
   protected Optional<Dag<JobExecutionPlan>> initialize(DagManagementStateStore dagManagementStateStore)
       throws IOException {
     try {
-      DagActionStore.DagAction dagAction = this.launchDagTask.getDagAction();
+      DagActionStore.DagAction dagAction = this.getDagTask().getDagAction();
```

Comment:
since I don't see the `DagProc` derived classes actually using the `DagTask` anywhere, it might be possible to be `private`.

then provide a `protected DagProc DagProc::getDagAction()` for these derived classes

Reply:
what about `@Getter(AccessLevel.PROTECTED)`?

Reply:
I don't yet detect a direct need for the `DagProc::getDagTask` method, even in derived classes.  all the latter are after amounts to "getDagAction"